### DISCUSSION TITLE
Disabled Models Still Appear While Enabled Models Disappear in Model Selection

### DISCUSSION THREAD

**Discussion by honwei189** [open-webui/open-webui#7912] [Issues]
**Description:**  
When I disable specific OpenRouter models (e.g., those with the same name but with a suffix like "(free)" or ":free") under **Admin -> Settings > Model**, I encounter an issue where the disabled models still appear in the model selection list when starting a new chat, while the enabled models disappear. (file attached below)

For example:  
- The model list contains two similar entries:  
   - `Meta: Llama 3.1 70B Instruct` (Model ID: `meta-llama/llama-3.1-70b-instruct`)  
   - `Meta: Llama 3.1 70B Instruct (free)` (Model ID: `meta-llama/llama-3.1-70b-instruct:free`)  
- If I disable `Meta: Llama 3.1 70B Instruct`, and then open a new chat, the model list **incorrectly displays** `Meta: Llama 3.1 70B Instruct` (the disabled model) while **omitting** `Meta: Llama 3.1 70B Instruct (free)` (the enabled model).  

---

**Steps to Reproduce:**  
1. Go to **Admin -> Settings > Model**.  
2. Disable a model with a suffix-free name, such as `Meta: Llama 3.1 70B Instruct`.  
3. Leave enabled its variant, e.g., `Meta: Llama 3.1 70B Instruct (free)`.  
4. Open a new chat and attempt to select a model.  

**Expected Behavior:**  
- The disabled model (`Meta: Llama 3.1 70B Instruct`) should not appear in the model selection list.  
- The enabled model (`Meta: Llama 3.1 70B Instruct (free)`) should appear in the model selection list.  

**Actual Behavior:**  
- The disabled model (`Meta: Llama 3.1 70B Instruct`) incorrectly appears in the list.  
- The enabled model (`Meta: Llama 3.1 70B Instruct (free)`) does not appear.  

---

**Additional Context:**  
Please see the attached screenshot for a visual example of the issue.  

This bug causes confusion when selecting models, particularly when managing large lists with similarly named entries. Resolving this issue will significantly improve model management usability.  

---

**Attachment(s):**  
![2024-12-17_085717](https://github.com/user-attachments/assets/ba1f752a-7e36-4788-ae1d-a044feb24cfe)
![2024-12-17_085810](https://github.com/user-attachments/assets/395e5d5f-86f3-421c-abcd-19e719bf68f5)

**Comment by honwei189** [comment]
Hi, is that any finding or update?

**Comment by carstenrossi** [comment]
Same here. Model is disabled, but shown in the frontend UI.

**Comment by jamius19** [comment]
I'm also facing the same issue when using OpenRouter API

**Comment by vuxngxn** [comment]
I fixed it by enable and disable the model again.
